test/code/plugins/oms_common_test.rb

```diff
@@ -94,6 +100,256 @@ def teardown
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "1.2.3.255"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.168.0.256"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "255.255.255.255"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.168.0"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.168.5."
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192..168.5"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.168..5"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192..168.9.5"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.168..9.5"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.168.9..5"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.168.255"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.168.5.6.7"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+      s = "192.A.5.6.7"
+      r = Common.is_like_ipv4_string?(s)
+      assert(!r,"Common.is_like_ipv4_string?(#{s}) should NOT be true.")
+    end
+
+    def test_is_like_ipv6_string?
+      s = "2001:0db8:85a3:0000:0000:8a2e:0370:7334"
+      r = Common.is_like_ipv6_string?(s)
+      assert(r,"Common.is_like_ipv6_string?(#{s}) should be true.")
+      s = "1234:DDDD:FFFF:EEEE:3333:8888:7777:1111"
+      r = Common.is_like_ipv6_string?(s)
+      assert(r,"Common.is_like_ipv6_string?(#{s}) should be true.")
+
+      s = "XXXX:0db8:85a3:0000:0000:8a2e:0370:7334"
+      r = Common.is_like_ipv6_string?(s)
+      assert(!r,"Common.is_like_ipv6_string?(#{s}) should NOT be true.")
+      s = "xx:0db8:85a3:0000:0000:8a2e:0370:7334"
+      r = Common.is_like_ipv6_string?(s)
+      assert(!r,"Common.is_like_ipv6_string?(#{s}) should NOT be true.")
+      s = ":0db8:85a3:0000:0000:8a2e:0370:7334"
+      r = Common.is_like_ipv6_string?(s)
+      assert(!r,"Common.is_like_ipv6_string?(#{s}) should NOT be true.")
+      s = "0db8:85a3:0000:0000:8a2e:0370:7334"
+      r = Common.is_like_ipv6_string?(s)
+      assert(!r,"Common.is_like_ipv6_string?(#{s}) should NOT be true.")
```

Comment:
Each of these sections of 3 lines should be reduced as much as possible; ideally into one line for each assert, since you're doing so many of them. I would recommend making an interface for each of the methods you're trying to test, like these system tests: https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/test/installer/scripts/omsadmin_systest.rb#L214
Especially for these, a `should_succeed` parameter can be very handy. Remember, if the test fails, it will print out something like "r was expected to be true, but was false", so having a message like "it should NOT be true" is redundant. Instead, both provide a useful message to the tester and document the reason for the test by saying something like "This IPv6 address is invalid because xxxx"

Reply:
Okay, I didn't see this last until just now, and I cleaned it up the way I'm used to seeing it.  I'll keep this pattern in mind though, as I recall we are doing that elsewhere.